ui/components/multichain/network-list-menu/network-list-menu.tsx

```diff
@@ -494,7 +495,7 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
         <AddRpcUrlModal
           onAdded={(url, name) => {
             // Note: We could choose to rename the URL if it already exists with a different name
-            if (rpcUrls.rpcEndpoints?.every((e) => e.url !== url)) {
+            if (rpcUrls.rpcEndpoints?.every((e) => !URI.equal(e.url, url))) {
```

Comment:
This matches the validation done by the network controller, which does not allow duplicates